cedar-policy-core/src/evaluator.rs

```diff
@@ -393,11 +393,41 @@ impl<'e> Evaluator<'e> {
                 match op {
                     BinaryOp::Eq => Ok((arg1 == arg2).into()),
                     // comparison and arithmetic operators, which only work on Longs
-                    BinaryOp::Less
-                    | BinaryOp::LessEq
-                    | BinaryOp::Add
-                    | BinaryOp::Sub
-                    | BinaryOp::Mul => {
+                    BinaryOp::Less | BinaryOp::LessEq => {
+                        let long_op = if matches!(op, BinaryOp::Less) {
+                            |x, y| x < y
+                        } else {
+                            |x, y| x <= y
+                        };
+                        let ext_op = if matches!(op, BinaryOp::Less) {
+                            |x, y| x < y
+                        } else {
+                            |x, y| x <= y
+                        };
+                        match (arg1.value_kind(), arg2.value_kind()) {
+                            (
+                                ValueKind::Lit(Literal::Long(x)),
+                                ValueKind::Lit(Literal::Long(y)),
+                            ) => Ok(long_op(x, y).into()),
+                            (ValueKind::ExtensionValue(x), ValueKind::ExtensionValue(y))
+                                if x.supports_operator_overloading()
+                                    && y.supports_operator_overloading()
+                                    && x.typename() == y.typename() =>
+                            {
+                                Ok(ext_op(x, y).into())
+                            }
+                            // throw type errors
+                            (ValueKind::Lit(Literal::Long(_)), _) => Err(EvaluationError::type_error_single(Type::Long, &arg2)),
+                            (_, ValueKind::Lit(Literal::Long(_))) => Err(EvaluationError::type_error_single(Type::Long, &arg1)),
+                            (ValueKind::ExtensionValue(x), _) if x.supports_operator_overloading() => Err(EvaluationError::type_error_single(Type::Extension { name: x.typename() }, &arg2)),
```

Comment:
I thought about it but abandoned this idea because adding a function will slow down the evaluator. We can mark it with `#[inline]` but I'm not certain it will be always enforced.